### DISCUSSION TITLE
Uploading a json file into Supabase storage

### DISCUSSION THREAD

**Discussion by marypaulm** [supabase/supabase#13979] [Questions]
Hello everyone,i need help on how to upload a json file into a storage,i could do it with a table but my use case needs storage how can i do that please,i tried using the bucket file path but nthing is working for days now,also tryied seraching for such an issue but still i didnt get answers.Kindly someone explain

**Comment by GaryAustin1** [comment]
Are you sure this has anything to do with JSON file and not setting RLS policies on your storage.object table to allow inserts?
Do image file uploads work?
You don't show how you are calling or anything about your storage policies, so hard to help.

**Reply by marypaulm** [reply]
Hi @GaryAustin1 thanks here are more details...so i have this sample file that i have named testingroms.json with data(shown below).
i tried saving it by the code hereby..am using supabase in c#..I have created a bucket called roomsdata,so am trying to upload the testingrooms json file but am new to supabase and i didnt find really much please explain how i can achieve this and if i also want to add more info abt the file like date created like if it were ina table how do i go about it?Thanks
I have aready declared the supabase client and provided the necessary url and api and its working when i upload on a table,also at the moment i have disabled Row security
 var response = await supabase.Storage
                .From("roomsdata")
                .Upload("testingrooms.json", rooms_json);

**Reply by GaryAustin1** [reply]
Your code seems to match the docs for c#.  I don't use it though.
I'm not sure how you can disable RLS on storage.objects.
You have to have an insert policy for storage.objects to upload. 
What is your error?
https://supabase.com/docs/guides/storage/access-control has info on policies.  There is also a policy area in the dashboard.

**Comment by isahir0711** [comment]
I dont, know if you still have the issue, but I found a way to solve it, go to your bucket policys, and create a policy for it, in this example is an Allow all, but you can customize it later depend on your needings
![image](https://github.com/supabase/supabase/assets/56800881/4b45fe4f-3de4-43e5-ba09-4f0dae5f6ee9)

**Reply by BobZiyangDing** [reply]
I think I solved it.

When uploading json, make sure you add this .encode("utf-8"), without it it wont work. 

    response = supabase.storage.from_("bucket_name").upload(
        path=file_address,
        file=json_data.encode("utf-8"),
        file_options={"cache-control": "3600", "upsert": "false"},
    )
